### PR TITLE
Pass coverage builds through task queue

### PR DESCRIPTION
**Pull request by danlamanna**
@mgrauer @zackgalbreath @cpatrick

### CODE REVIEW

**Comment by danlamanna** [comment]
@zackgalbreath This is ready for review.
The only immediate shortcoming I know needs to be fixed is the similar fixing of `sys_get_temp_dir` within the other 2 handlers using it.

**Comment by danlamanna** [comment]
@zackgalbreath @mgrauer

This is ready for review. Unfortunately, due to time constraints this feature is disabled by default and marked experimental since there are issues with the consumer-based tests running. (Zack, this may be related to the issue we discussed about consumer configuration staying in memory).

Nonetheless, the existing test suite passes and this code (with a different consumer) has been working on CDash GAE for quite some time and this PR is an attempt to make deploying for that use case easier (rather than maintaining a separate branch forever).

**Comment by mgrauer** [comment]
I definitely don't want to have to maintain a separate branch.

File: public/api/v1/getSubmissionFile.php [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+  Copyright (c) Kitware, Inc. All rights reserved.
+  See LICENSE or http://www.cdash.org/licensing/ for details.
+
+  This software is distributed WITHOUT ANY WARRANTY; without even
+  the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+  PURPOSE. See the above copyright notices for more information.
+=========================================================================*/
+
+$noforcelogin = 1;
```
Comment (zackgalbreath) [inline public/api/v1/getSubmissionFile.php]: $noforcelogin is only needed when including login.php

File: config/config.php [hunk 2]
```diff
@@ -74,6 +74,9 @@
 $CDASH_BERNARD_DRIVER = false;
 $CDASH_BERNARD_CONSUMERS_WHITELIST = false;
 
+// EXPERIMENTAL: Whether or not to use Bernard for submitting coverage jobs
+$CDASH_BERNARD_COVERAGE_SUBMISSION = false;
```
Comment (zackgalbreath) [inline config/config.php]: What happens when `$CDASH_BERNARD_SUBMISSION` is true and `$CDASH_BERNARD_COVERAGE_SUBMISSION` is false?  I guess coverage builds fall back to one of our older processing pipelines based on whether or not `$CDASH_ASYNCHRONOUS_SUBMISSION` is set?

Comment (danlamanna) [inline config/config.php]: Correct. For now it's `$CDASH_BERNARD_SUBMISSION` corresponds to non-coverage submissions and `$CDASH_BERNARD_COVERAGE_SUBMISSION` corresponds to coverage-only submissions.

**Comment by danlamanna** [comment]
Updated.

**Pull request merged**